src/vdom/index.js

```diff
@@ -36,6 +36,7 @@ export function isNamedNode(node, nodeName) {
 export function getNodeProps(vnode) {
 	let props = extend({}, vnode.attributes);
 	props.children = vnode.children;
+	if (props.children === undefined) delete props.children;
```

Comment:
I think a better way would be to put the condition around the line above:

```js
if (vnode.children !== undefined) {
  props.children = vnode.children;
}
```

The `delete` keyword is known to cause a deopt in V8.